File: CA4TJ/ca4.py

import math

# Addition

add = lambda x,y: x+y

x = (2, 2, -2, 2, 3, 0.5)
y = (2, 4, 2, -2, 0, 0.5)
adding = map(add, x, y)
print adding
print round(add(2/4.0, 2/6.0), 4)

# Addition by reduce function

addbyred = reduce(lambda x, y: x + y, [1, 2, 3, 4])
print addbyred

# Addition by mapping 3 sets

a = [1, 2, 3, 4]
b = [5, 6, 7, 8]
c = [-2, -4, -6, -8]

map (lambda x, y, z: x + y + z, a, b, c)
print map (lambda x, y, z: x + y + z, a, b, c)


# Subtraction

subtract = lambda x,y: x-y

x = (2, 3, 2, -2, -2, 0.5)
y = (2, 2, 3, -3, 2,  0.2)
sub = map(subtract, x, y)
print sub 
print round(subtract(2/4.0, 2/6.0), 4) 

# Subtraction by reduce function

subbyred = reduce(lambda x, y: x - y, [1, 2, 3, 4])
print subbyred

# Subtraction by mapping 3 sets
map (lambda x, y, z: x + y - z, a, b, c)
print map (lambda x, y, z: x + y - z, a, b, c)


# Multiplication

multiply = lambda x,y: x*y
x = (2, 0, 2, -2, 2, -2, 0.5)
y = (2, 2, 0, 0, -2, -2, 0.5)
mult = map(multiply, x, y)
print mult
print round(multiply(2/4.0, 2/6.0), 4)

# Multiplication by reduce function

multbyred = reduce(lambda x, y: x * y, [1, 2, 3, 4])
print multbyred

# Multiplication by mapping 3 sets

map (lambda x, y, z: x * y * z, a, b, c)
print map (lambda x, y, z: x * y * z, a, b, c)


# Division

def divide(x, y):
	if y == 0:
		print "Can't divide by zero"
	else:
		return x/y

x = (6, -6, 2, 16, 0, 4)
y = (3, 3, 2, 8, 4, 0)
div = map(divide, x,y)
print div

# Division by reduce function

divbyred = reduce(lambda x, y: x / y, [24, 2, 2, 2])
print divbyred

# Division by mapping 3 sets
a = (8, 27, 64, 125)
b = (2, 3, 4, 5)
c = (2, 3, 4, 5)

map (lambda x, y, z: x/y/z, a, b, c)
print map (lambda x, y, z: x/y/z, a, b, c)


# Factorial

def factorial(x):
	if x > 1:
		return x * factorial(x-1)
	if x < 0:
		return "inf"
	return 1
	
x = (-1, 0, 1, 3, 5, 10)
fact = map(factorial, x)
print fact


# Squaring a number

def squared(x):
	return x*x
	
x = (-1, 0, 1, 3, 5, 10, 0.5)
square = map(squared, x)
print square


# Squaring using a Generator

x = (x ** 2 for x in range(20))
x = list(x)
print (x)



# Cubed Numbers

def cubed(x):
	return x*x*x
	
x = (-1, 0, 1, 3, 5, 10, 0.5)
cube = map(cubed, x)
print cube

# Cubing using a Generator

x = (x ** 3 for x in range(10))
x = list(x)
print (x)


# Square root

def root(x):
	if x >= 0:
		return x**(0.5)
	if x < 0:
		return "imaginary number"
		
x = (4, 9, 0, -9, 2.25)
sroot = map(root, x)
print sroot
	
	
# Cubed root

def cubedroot(x):
	if x >= 0:
		return x**(1/3.0)
	if x < 0:
		return -(-x)**(1/3.0)
		
x = (8, -8, 0, 0.125)
croot = map(cubedroot, x)
print croot


# Exponent 

def exp (x, y):
	return x ** y
	
x = (0, 1, 1, 2, 5, -5)
y = (10, 0, 6, 5, -3, 3)
exponent = map(exp, x, y)
print exponent


# Radians, Convert degrees to radians

def rad (x):
	return (x*math.pi)/180
	
x = (50, 100, -10, 0, 57.2958)
radians = map(rad, x)
print radians


# Sine

def sine (x):
	return math.sin(x)
	
x = (100, 90, 250, -50)
Sine = map(sine, x)
print Sine


# Cosine

def cos (x):
	return  math.cos(x)
	
x = (90, 100, 190, 300, -50)
Cos = map(cos, x)
print Cos


# Tangent

def tan (x):
	return math.tan(x)
	
x = (45, 100, 300, -10)
tangent = map(tan, x)
print tangent


# Celsius and Fahrenheit

Celsius = [-17.8, -10, -5, 0, 10, 20, 50]
Fahrenheit = map(lambda x: (float(9)/5)*x + 32, Celsius)
print Fahrenheit

C = map(lambda x: (float(5)/9)*(x-32), Fahrenheit)
print C

File: CA4TJ/CA 4.R

# Addition

add <- function(x, y)
{return (paste("The sum of the two numbers is", (x + y)))}
  
# test addition

add(2, 2)
add(2, 4)
add(-2, 2)
add(2, -2)
add(4, -1)
add(3, 0)
add(0.5, 0.3)

# Subtraction

subtract <- function(x, y)
{return (paste("The result of", x, "minus", y, "is", (x - y)))}

# test subtraction

subtract(2, 2)
subtract(3, 2)
subtract(2, 3)
subtract(-2, -3)
subtract(-2, 2)
subtract(0.5, 0.4)

# Multiplication

multiply <- function(x, y)
{return (paste("The product of the two numbers is", (x*y)))}

# test multiplication

multiply(2, 2)
multiply(2, 3)
multiply(0, 2)
multiply(2, 0)
multiply(-2, 0)
multiply(2, -2)
multiply(-2, -2)
multiply(0.5, 0.25)
  
# Division

divide <- function(x, y)
{return(paste("The result of", x, "divided by", y, "is", (x/y)))}

# test division

divide (6, 3)
divide (-6, 3)
divide (0.5, 0.5)
divide (0, 4)
divide (4, 0)
divide (0.5, 0.25)
divide (0.5, -0.25)


# Factorial

fact <- function(x)
{return(paste("The factorial of", x, "is", (factorial(x))))}

# test factorial

fact(5)
fact(0)
fact(1)
fact(6)
fact(10)
fact(-3)


# Squared

squared <- function(x)
{return(paste(x, "squared is", (x*x)))}

# test squared

squared(10)
squared(-10)
squared(0)
squared(0.25)
squared(-1.25)


# Cubed

cubed <- function(x)
{return(paste(x, "cubed is", (x*x*x)))}

# test cubed 

cubed(10)
cubed(-10)
cubed(0)
cubed(0.5)
cubed(-0.5)
cubed(1.25)


# Square Root

root <- function(x)
{return(paste("The square root of", x, "is", (x**0.5)))}

# test square root

root(9)
root(5)
root(-9)
root(0)


# Exponent

exp <- function(x, y)
{return(paste(x, "to the power of", y, "is", (x**y)))}

# test exponent

exp(2,9)
exp(2, -9)
exp(-2, -9)
exp(0.3,4)
exp(0.3, 0.2)
exp(0.3, -0.2)


# Sine 

sine <- function(x)
{return(paste("The sine of", x, "is", (sin(x)), "radians"))}

# test sine

sine(100)
sine(90)
sine(250)
sine(-50)


# Cosine

cosine <- function(y)
{return(paste("The cosine of", y, "is", (cos(y)), "radians"))}

#  test cos

cos(90)
cos(100)
cos(190)
cos(300)
cos(-50)


# Tangent

tangent <- function(x)
{result(paste("The tan of", x, "is", (tan(x)), "radians"))}

# test tan

tan(45)
tan(100)
tan(300)
tan(-10)

